Library/Homebrew/test/tab_test.rb

```diff
@@ -69,6 +69,53 @@ def test_universal?
     assert_predicate tab, :universal?
   end
 
+  def test_homebrew_tag
+    tab = Tab.new(homebrew_version: "1.2.3")
+    assert_equal "1.2.3", tab.homebrew_tag
+
+    tab.homebrew_version = "1.2.4-567-g12789abdf"
+    assert_equal "1.2.4", tab.homebrew_tag
```

Comment:
I think it's worth testing these are greater or less rather than the actual value matches so we may be able to strip the modification logic.